src/redux/Courier/taskEntityReducer.js

```diff
@@ -81,6 +84,17 @@ function replaceItem(state, payload) {
   return state;
 }
 
+
+function updateItem(state, id, payload) {
+  const index = state.findIndex(item => item['@id'] === id);
+
+  if (index !== -1) {
+    return state.map((item, i) => (i === index ? { ...item, ...payload } : item));
```

Comment:
why `map`? can be just `state[index] = { ...state[index], ...payload };` ?

Reply:
For immutability, objects in javascript are passed via reference. So i should not edit the `state` variable.

Reply:
You'd to clone the object in order to use your snippet
```javascript
const updatedItems = [...prevItems]; // Create a copy of the array
updatedItems[index] = { ...updatedItems[index], ...payload };
```

It's possible that way, but i'm more used to functional paradigm